From f9cf5437e57fd657a7aa3851b50c879e096659cc Mon Sep 17 00:00:00 2001
From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 29 Apr 2014 17:07:40 -0400
Subject: [scripts] headers_check: special case seqbuf_dump()

Message-id: <1398791260-24689-4-git-send-email-prarit@redhat.com>
Patchwork-id: 79230
O-Subject: [RHEL7.1 PATCH BZ 1090078 3/3] headers_check: special case seqbuf_dump()
Bugzilla: 1090078
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>
RH-Acked-by: Rik van Riel <riel@redhat.com>

Bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1090078

commit a7e1d98f3e2a0d858fddcac7c66b78b6dcfd9d2e
Author: Paul Bolle <pebolle@tiscali.nl>
Date:   Thu Jan 23 15:54:08 2014 -0800

    headers_check: special case seqbuf_dump()

    "make headers_check" warns about soundcard.h for (at least) five years
    now:
        [...]/usr/include/linux/soundcard.h:1054: userspace cannot reference function or variable defined in the kernel

    We're apparently stuck with providing OSSlib-3.8 compatibility, so let's
    special case this declaration just to silence it.

    Notes:

    0) Support for OSSlib post 3.8 was already removed in commit 43a990765a
       ("sound: Remove OSSlib stuff from linux/soundcard.h").  Five years have
       passed since that commit: do people still care about OSSlib-3.8?  If
       not, quite a bit of code could be remove from soundcard.h (and probably
       ultrasound.h).

    2) By the way, what is actually meant by:
        It is no longer possible to actually link against OSSlib with this
        header, but we still provide these macros for programs using them.

    Doesn't that mean compatibility to OSSlib isn't even useful?

    3) Anyhow, a previous discussion soundcard.h, which led to that commit,
       starts at https://lkml.org/lkml/2009/1/20/349 .

    4) And, yes, I sneaked in a whitespace fix.

    Signed-off-by: Paul Bolle <pebolle@tiscali.nl>
    Cc: Takashi Iwai <tiwai@suse.de>
    Acked-by: Arnd Bergmann <arnd@arndb.de>
    Cc: Michal Marek <mmarek@suse.cz>
    Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
    Signed-off-by: Linus Torvalds <torvalds@linux-foundation.org>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/scripts/headers_check.pl b/scripts/headers_check.pl
index 64ac238..62320f9 100644
--- a/scripts/headers_check.pl
+++ b/scripts/headers_check.pl
@@ -65,7 +65,11 @@ sub check_include
 
 sub check_declarations
 {
- if ($line =~m/^(\s*extern|unsigned|char|short|int|long|void)\b/) {
+ # soundcard.h is what it is
+ if ($line =~ m/^void seqbuf_dump\(void\);/) {
+  return;
+ }
+ if ($line =~ m/^(\s*extern|unsigned|char|short|int|long|void)\b/) {
   printf STDERR "$filename:$lineno: " .
          "userspace cannot reference function or " .
          "variable defined in the kernel\n";
-- 
1.7.1